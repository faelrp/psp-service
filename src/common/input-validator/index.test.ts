import * as yup from 'yup';
import inputValidator from './index';

describe('Schema validator', () => {
  test('should not be valid if input not respect the field type', () => {
    const schema = yup.object().shape({
      test: yup.string(),
    });
    const payload = { test: 123 };

    expect(inputValidator.validate(schema, payload)).toMatchSnapshot();
  });

  test('should not be valid if required field is missing', () => {
    const schema = yup.object().shape({
      test: yup.string().required(),
    });
    const payload = {};

    expect(inputValidator.validate(schema, payload)).toMatchSnapshot();
  });

  test('should return null if optional field is missing', () => {
    const schema = yup.object().shape({
      test: yup.string(),
    });
    const payload = {};

    expect(inputValidator.validate(schema, payload)).toMatchSnapshot();
  });
});
