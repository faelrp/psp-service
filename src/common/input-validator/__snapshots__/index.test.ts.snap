// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema validator should not be valid if input not respect the field type 1`] = `[ValidationError: test must be a \`string\` type, but the final value was: \`123\`.]`;

exports[`Schema validator should not be valid if required field is missing 1`] = `[ValidationError: test is a required field]`;

exports[`Schema validator should return null if optional field is missing 1`] = `null`;
